test/functional/p2p_ibd.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test inv messages and fee filters during and after IBD."""
+
+from decimal import Decimal
+
+from test_framework.messages import COIN
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal
+
+MAX_FEE_FILTER = Decimal(str(9170997 / COIN))
+NORMAL_FEE_FILTER = Decimal(str(100 / COIN))
+
+
+class P2PIBDTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 2
+        self.extra_args = [["-minrelaytxfee=0.00000100"], ["-minrelaytxfee=0.00000100"]]
+
+    def run_test(self):
+        self.log.info("Check that nodes set minfilter to MAX_MONEY and don't receive invs while still in IBD")
+        for node in self.nodes:
+            assert node.getblockchaininfo()['initialblockdownload']
+            info = node.getpeerinfo()
+            for conn_info in info:
+                assert_equal(conn_info['minfeefilter'], MAX_FEE_FILTER)
+                assert "inv" not in conn_info["bytesrecv_per_msg"].keys()
+                assert "inv" not in conn_info["bytessent_per_msg"].keys()
+
+        # Come out of IBD by generating a block
+        self.nodes[0].generate(1)
+        self.sync_all()
+
+        self.log.info("Check that nodes reset minfilter and receive invs after coming out of IBD")
+        for node in self.nodes:
+            assert not node.getblockchaininfo()['initialblockdownload']
+            info = node.getpeerinfo()
+            for conn_info in info:
+                assert_equal(info[0]['minfeefilter'], NORMAL_FEE_FILTER)
+                assert "inv" in conn_info["bytesrecv_per_msg"].keys() or "inv" in conn_info["bytessent_per_msg"].keys()
```

Comment:
This is a inv of type block, no?

I am not sure if there is a way to ask Bitcoin Core which inv type it received. Maybe the only way is to use mininode?

Reply:
Well, to be honest, this test doesn't even produce any transactions to send invs about right now 😅  I need to work that out.

Mininodes are useful for seeing what nodes send out, but I'm not sure how useful they for seeing what nodes receive... I could have a mininode manually send a feefilter and make sure it doesn't receive anything, but then I'm testing the feefilter message instead of IBD behavior.

Reply:
Isn't the point to test what the node sends out? I.e. test whether MAX_FILER or not was sent out?

Reply:
Yep, feefilter is already tested - `MAX_FILTER` during IBD, then reset to normal after.
I had been hoping to also test txrelay, but it seems more complicated than what I was doing originally.

Reply:
@MarcoFalke if you have time to check out https://github.com/bitcoin/bitcoin/pull/19423/commits/7bc653e7e3d21c675c6a1257f658222b97e166cb please let me know what you think :)